### PR TITLE
WIP: validate against boydorr/rdiversity R package

### PR DESCRIPTION
**Pull request by richardreeve**
Our main codebase for diversity measurement is currently in R - the github repository [boydorr/rdiversity](https://github.com/boydorr/rdiversity). This PR validates the code we have here against that package.

### CODE REVIEW

**Comment by tkelman** [comment]
What happens if someone doesn't have R installed and tries to run `Pkg.test("Diversity")` now? There are still a fair number of tests, aside from the R validation, that could work, right?

File: test/pkg_RCall.jl [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+# Only run R on macs
+if is_apple()
+    # Environment variable to avoid boring R package builds
+    is_me = haskey(ENV, "SKIP_R_INSTALL") && ENV["SKIP_R_INSTALL"] == "1"
+    # Skip the (slow!) R package installation step
+    if is_me
+        R"library(rdiversity)";
+    else
+        libdir = mktempdir();
```
Comment (tkelman) [inline test/pkg_RCall.jl]: would you want to clean this up when the tests are over, if it's going to be different every time anyway?

**Comment by richardreeve** [comment]
If you don't have R installed, then RCall will install, but `using RCall` will fail. I don't know how to get around that - there is a discussion at JuliaInterop/RCall.jl#73 about automatic installation of R if it's missing, but I don't see how to check for this in the test.

File: test/pkg_RCall.jl (hunk 1 above)
Comment (richardreeve) [inline test/pkg_RCall.jl]: It creates a folder in /tmp, so I was relying on the OS to clean it up. However, I've now fixed this in master.

**Comment by richardreeve** [comment]
Incidentally, this is a particular problem because when RCall fails to load, the R_str macro is not defined and the file load fails...

**Comment by richardreeve** [comment]
Okay, I seem to have fixed this by avoiding macros and using a `try` block.

File: test/pkg_RCall.jl (hunk 1 above)
Comment (tkelman) [inline test/pkg_RCall.jl]: not all os'es clean up the temp dir contents

**Pull request merged**